stock_prices/stock_prices.py

```diff
@@ -3,7 +3,29 @@
 import argparse
 
 def find_max_profit(prices):
-  pass
+    #Declare "smallest" equal to the first element in the given array
+    smallest = prices[0]
+    #For each index in the array
+    for i in range(len(prices)):
+        #if that index is less than the previously set smallest value AND that index does not equal the same value of the last element in prices
+        if prices[i] < smallest and prices[i] != prices[len(prices) - 1]:
+            #Then "smallest" equals that index
+            smallest = prices[i]
+    
+    #"Smallest_value" equals the smallest index value in prices
+    smallest_value_index = prices.index(smallest)
+    #"shortened_prices" equals the element within prices thats just 1 higher in the array
+    shortened_prices = prices[smallest_value_index + 1 :]
+    largest = shortened_prices[0]
+
```

Comment:
Good comments, you don't have to write every line, and more focus on breaking the logic of each conditional or loops.